CMakeLists.txt

```diff
@@ -1,14 +1,59 @@
 cmake_minimum_required(VERSION 3.0)
 
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.14.12.tar.gz"
+    SHA1 "2d444140db0d53393b5490604006bb4f4896a444"
+)
+
+option(ENABLE_CPP11 "Build websocketpp with CPP11 features enabled." TRUE)
+
 project(websocketpp VERSION 0.7.0)
 
 set(include_install_dir "include")
 set(config_install_dir "lib/cmake/${PROJECT_NAME}")
 
+hunter_add_package(Boost COMPONENTS chrono regex system random thread)
+hunter_add_package(OpenSSL)
```

Comment:
Since all dependencies of cacheable package should be cacheable too we need to make OpenSSL cacheable first. We need to add extra step to OpenSSL build scheme to install `OpenSSLConfig.cmake` with imported targets. See Boost for example:
* https://github.com/ruslo/hunter/blob/14aeeb2f41b77ac989f18d90e8df82b3b7bc0d32/cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in#L369